### PR TITLE
Let textareas resize automatically without having to highlight placeholders

### PR DESCRIPTION
**Pull request by quis**
Scrolling within `<textarea>`s on the page is a bit grim. Which is why we don’t do it for the textboxes that people use to edit templates.

This pull request extends the auto-resizing of `<textarea>`s to those which don’t need the highlighting of placeholders.

***

awanp

***

image

### CODE REVIEW

File: app/assets/javascripts/enhancedTextbox.js [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+        )
+      );
+
+      if ('stickAtBottomWhenScrolling' in GOVUK) {
+        GOVUK.stickAtBottomWhenScrolling.recalculate();
+      }
+
+    };
+
+    this.contentEscaped = () => $('<div/>').text(this.$textbox.val()).html();
```
Comment (tombye) [inline app/assets/javascripts/enhancedTextbox.js]: Is the content escaped here? I can't find reference to it in the docs for `.val()`.

Comment (quis) [reply app/assets/javascripts/enhancedTextbox.js]: There some detail in this commit: https://github.com/alphagov/notifications-admin/commit/836fe77c372eaa10bf6273b0efef54f6bcd709bb

TL;DR is that it’s the `$('<div/>').text(…)` part that does the escaping.

File: app/assets/javascripts/enhancedTextbox.js [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+  const tagPattern = /\(\(([^\)\((\?)]+)(\?\?)?([^\)\(]*)\)\)/g;
+
+  Modules.EnhancedTextbox = function() {
+
+    this.start = function(textarea) {
+
+      let visibleTextbox;
+
+      this.highlightPlaceholders = (
+        typeof textarea.data('highlightPlaceholders') === 'undefined' ||
```
[the remark this answers is not in the document]
Comment (quis) [reply app/assets/javascripts/enhancedTextbox.js]: `this.highlightPlaceholders` should be `true` if:
- `data-highlightPlaceholders` is explicitly set to `"true"`
- `data-highlightPlaceholders` is not set (i.e. it’s true by default)

File: app/assets/javascripts/enhancedTextbox.js (hunk 1 above)
Comment (tombye) [reply app/assets/javascripts/enhancedTextbox.js]: Gotcha, sorry, should have checked those docs.

File: app/assets/javascripts/enhancedTextbox.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (tombye) [reply app/assets/javascripts/enhancedTextbox.js]: Right, I didn't know the second point but see there's a test for it so 👍

**Comment by quis** [comment]
@tombye I fixed the calls to the textbox macro, and added a test to make sure the right data attributes are present on the edit templates page: https://github.com/alphagov/notifications-admin/compare/resize-all-textbox-dem#diff-f2eb304b93cc383727c0ab7fc8fbd464R349-R350
